### PR TITLE
Rework SimpleBusyLoop, MultiSharedMultipleRegions, and SimpleFeatureInteraction

### PR DESCRIPTION
**Pull request by jonas-kaufmann**
- Removed SimpleSleepLoop because it will context switch away to perform the sleep, which leads to high run-to-run variance.
- Added the feature variable annotations directly in code to avoid having to update the feature model for every change.
- SFI, MSMR, and SFI now use busy sleeping to make the feature regions appear in the perf sampling profile.

@vulder The commits where I modified SSL are still there. Should I remove them using a rebase?

### CODE REVIEW

**Comment by vulder** [comment]
No, I'll squash merge anyhow

File: src/MultiSharedMultipleRegions/MSMRmain.cpp [hunk 1]
```diff
@@ -27,29 +27,29 @@ int main(int argc, char *argv[] ) {
   // Multiple regions related to --slow that take different amounts of time.
 
   if (Slow) {
-    fpcsc::sleep_for_secs(5);
+    fpcsc::busy_sleep_for_secs(5);
   } else {
-    fpcsc::sleep_for_secs(3);
+    fpcsc::busy_sleep_for_secs(3);
```
Comment (vulder) [inline src/MultiSharedMultipleRegions/MSMRmain.cpp]: I like the idea of having a busy_sleep version that perf can detect. It just conflicts a bit with debugability. Would you be so nice an maybe add an abstraction where we can, with a compile switch, use busy VS normal sleeping?

File: include/fpcsc/perf_util/sleep.h [hunk 2]
```diff
@@ -12,15 +12,26 @@ inline void sleep_for_secs(unsigned Secs) {
...
+}
+
 inline void sleep_for_millisecs(unsigned Millisecs) {
   std::cout << "Sleeping for " << Millisecs << " milliseconds" << std::endl;
   std::this_thread::sleep_for(std::chrono::milliseconds(Millisecs));
 }
-
-inline void sleep_for_nanosecs(unsigned millisecs) {
-  std::this_thread::sleep_for(std::chrono::nanoseconds(millisecs));
-}
```
Comment (vulder) [inline include/fpcsc/perf_util/sleep.h:22]: Just, keep the function, maybe someone else will need it.

**Review by vulder — CHANGES_REQUESTED** [review]

File: include/fpcsc/perf_util/sleep.h (hunk 2 above)
Comment (jonas-kaufmann) [reply include/fpcsc/perf_util/sleep.h:22]: In SimpleSleepLoop, this lead to a lot of variance. I would rather force whoever plans to use it to carefully consider whether sleeping for a few nanoseconds is actually necessary as this corresponds to only a few hundred processor cycles. In my opinion, it makes sense to implement a busy sleep version instead if necessary.

**Review by vulder — APPROVED** [review]

File: include/fpcsc/perf_util/sleep.h (hunk 2 above)
Comment (vulder) [reply include/fpcsc/perf_util/sleep.h:22]: A busy version in addition would sure be nice. The core point about this is also not that it's super precise but more that one could, with the right experiment design, try to analyze implications to sampling resolution.
Now that I think about it, as perf does not find the sleep for this a busy version would be really practicable.

Comment (vulder) [reply include/fpcsc/perf_util/sleep.h:22]: Let's remove it for now and add one if we need it

**Pull request merged**